### DISCUSSION TITLE
How i can set custom icon for button?

### DISCUSSION THREAD

**Discussion by Whhhatttt** [flet-dev/flet#1619] [Q&A]

**Comment by Whhhatttt** [comment]
I tried to change the icons in the font, but it's a Google font. I do not understand where the addresses and the name are stored. How does the program understand that the "CUT" icon is the address 0x etc. How? I don't know how many days I've been struggling with this. In my app, users need to be able to upload their own icons. For in this library of icons there is absolutely nothing that I want, for example.

**Reply by stabldev** [reply]
If you just want to add custom icon for buttons, yes you can.
here is an example I did with `IconButton`,
you can use `content` prop for custom icon also [check this](https://flet.dev/docs/controls/iconbutton#content)
```py
ft.IconButton(
    content=ft.Image(
        src="icons/messenger-outline.svg",
        width=22,
        height=22,
        fit=ft.ImageFit.COVER,
        repeat=ft.ImageRepeat.NO_REPEAT,
    )
)

if __name__ == "__main__":
    ft.app(
        target=main,
        assets_dir="assets",
    )
```

**Reply by ndonkoHenri** [reply]
Or this example: https://github.com/flet-dev/examples/blob/main/python/controls/image/svg-lucide.py
